packages/app-mobile/setupQuickActions.ts

```diff
@@ -1,59 +1,47 @@
 	QuickActions.setShortcutItems([
 		{ type: 'New note', title: _('New note'), icon: 'Compose', userInfo },
 		{ type: 'New to-do', title: _('New to-do'), icon: 'Add', userInfo },
 	]);
-
-	const handleQuickAction = (data: TData) => {
-		if (!data) return;
-
-		// This dispatch is to momentarily go back to reset state, similar to what
-		// happens in onJoplinLinkClick_(). Easier to just go back, then go to the
-		// note since the Note screen doesn't handle reloading a different note.
-		//
-		// This hack is necessary because otherwise you get this problem:
-		// The first time you create a note from the quick-action menu, it works
-		// perfectly. But if you do it again immediately later, it re-opens the
-		// page to that first note you made rather than creating an entirely new
-		// note. If you navigate around enough (which I think changes the redux
-		// state sufficiently or something), then it'll work again.
-		dispatch({ type: 'NAV_BACK' });
-		dispatch({ type: 'SIDE_MENU_CLOSE' });
-
-		const isTodo = data.type === 'New to-do' ? 1 : 0;
-
-		void Note.save({
-			parent_id: folderId,
-			is_todo: isTodo,
-			// eslint-disable-next-line promise/prefer-await-to-then, @typescript-eslint/no-explicit-any -- Old code before rule was applied, Old code before rule was applied
-		}, { provisional: true }).then((newNote: any) => {
-			dispatch({
-				type: 'NAV_GO',
-				noteId: newNote.id,
-				folderId,
-				routeName: 'Note',
-			});
-		});
-	};
-
-	DeviceEventEmitter.addListener('quickActionShortcut', handleQuickAction);
-
-	// eslint-disable-next-line promise/prefer-await-to-then, @typescript-eslint/no-explicit-any -- Old code before rule was applied, Old code before rule was applied
-	QuickActions.popInitialAction().then(handleQuickAction).catch((reason: any) => reg.logger().error(reason));
 };
 
+export const quickActionHandler = async (data: TData, dispatch: Dispatch, selectedFolderId: string) => {
+	if (!data) return;
+
+	// This dispatch is to momentarily go back to reset state, similar to what
+	// happens in onJoplinLinkClick_(). Easier to just go back, then go to the
+	// note since the Note screen doesn't handle reloading a different note.
+	//
+	// This hack is necessary because otherwise you get this problem:
+	// The first time you create a note from the quick-action menu, it works
+	// perfectly. But if you do it again immediately later, it re-opens the
+	// page to that first note you made rather than creating an entirely new
+	// note. If you navigate around enough (which I think changes the redux
+	// state sufficiently or something), then it'll work again.
+	dispatch({ type: 'NAV_BACK' });
+	dispatch({ type: 'SIDE_MENU_CLOSE' });
+
+	const isTodo = data.type === 'New to-do' ? 1 : 0;
+
+	const newNote = await Note.save({
+		parent_id: selectedFolderId,
+		is_todo: isTodo,
+	}, { provisional: true });
+
+	dispatch({
+		type: 'NAV_GO',
+		noteId: newNote.id,
+		folderId: selectedFolderId,
+		routeName: 'Note',
+	});
```

Comment:
I wonder if it could make sense to use `CommandService.instance().execute('newNote', '', isTodo)` instead? This has the potential to be simpler, but edge cases would need to be tested (`newNote` is a relatively new command and is not well tested).

Reply:
I changed to the Command, retested everything and it seems to be working. The only changed that I saw in the code was that the command doesn't send the `folderId` on `NAV_GO`, which is used to set the `selectedFolderId`. Since this value is not used by `Note.tsx` anymore I guess we can just ignore it, right?